### PR TITLE
Add missing Transaction properties

### PR DESCRIPTION
**Pull request by truechernyshov**

### CODE REVIEW

File: src/Entities/Transaction.php [hunk 1]
```diff
@@ -725,4 +725,100 @@ public function getReferenceData()
...
+    {
+        return $this->getAttribute('hasBumpOffer');
+    }
+
+    /**
+     * @return string
+     */
+    public function getBumpOffer()
+    {
+        return $this->getAttribute('bumpOffer');
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:806]: That doesn't correct, it should be a object

File: src/Entities/Transaction.php [hunk 2]
```diff
@@ -725,4 +725,100 @@ public function getReferenceData()
...
+    {
+        return $this->getAttribute('3ds');
+    }
+
+    /**
+     * @return string
+     */
+    public function getDcc()
+    {
+        return $this->getAttribute('dcc');
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:750]: That doesn't correct, it should be a object

File: src/Entities/Transaction.php [hunk 3]
```diff
@@ -725,4 +725,100 @@ public function getReferenceData()
...
+    {
+        return $this->getAttribute('isRebill');
+    }
+
+    /**
+     * @return string
+     */
+    public function get3ds()
+    {
+        return $this->getAttribute('3ds');
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:742]: That doesn't correct, it should be an object

File: src/Entities/Transaction.php [hunk 4]
```diff
@@ -725,4 +725,100 @@ public function getReferenceData()
...
+    {
+        return $this->getAttribute('riskScore');
+    }
+
+    /**
+     * @return string
+     */
+    public function getRetryInstruction()
+    {
+        return $this->getAttribute('retryInstruction');
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:766]: This field might be deprecated soon, need to check.

File: src/Entities/Transaction.php (hunk 1 above)
Comment (truechernyshov) [reply src/Entities/Transaction.php:806]: Only return type or you mean should create bumpOffer entity before?

Comment (ashkarpetin) [reply src/Entities/Transaction.php:806]: Yes, it should create and return entity

Comment (truechernyshov) [reply src/Entities/Transaction.php]: Removed this getter.

File: src/Entities/Transaction.php (hunk 2 above)
Comment (truechernyshov) [reply src/Entities/Transaction.php]: Removed this getter.

File: src/Entities/Transaction.php (hunk 4 above)
Comment (truechernyshov) [reply src/Entities/Transaction.php]: Removed this getter.

File: src/Entities/Transaction.php (hunk 3 above)
Comment (truechernyshov) [reply src/Entities/Transaction.php:743]: Added ThreeDSecureResult

File: src/Entities/Transactions/ThreeDSecureResult.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * @see https://www.rebilly.com
+ */
+
+namespace Rebilly\Entities\Transactions;
+
+use Rebilly\Rest\Entity;
+
+/**
+ * Class ThreeDSecure.
```
Comment (ashkarpetin) [inline src/Entities/Transactions/ThreeDSecureResult.php:17]:
```suggestion
 * Class ThreeDSecureResult.
```

File: src/Entities/Transaction.php (hunk 3 above)
Comment (ashkarpetin) [reply src/Entities/Transaction.php:743]: Attribute factort is missing, resource ThreeDSecureResult won't be created, e.g.:
https://github.com/Rebilly/rebilly-php/blob/c6b8890c6fadb2690f2b45c9af7077d68ecc5869/src/Entities/Transaction.php#L461-L469

File: src/Entities/Transaction.php [hunk 6]
```diff
@@ -459,6 +469,16 @@ public function createRiskMetadata(array $data)
         return new RiskMetadata($data);
     }
 
+    /**
+     * @param array $data
+     *
+     * @return ThreeDSecureResult
+     */
+    public function create3dc(array $data)
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:477]:
```suggestion
    public function create3ds(array $data)
```

File: src/Entities/Transaction.php [hunk 7]
```diff
@@ -449,6 +449,16 @@ public function setRiskMetadata(RiskMetadata $value)
...
 
+    /**
+     * @param ThreeDSecureResult $value
+     *
+     * @return $this
+     */
+    public function set3ds(ThreeDSecureResult $value)
+    {
+        return $this->setAttribute('3ds', $value);
+    }
```
Comment (ashkarpetin) [inline src/Entities/Transaction.php:460]: Setter is not needed, it's a read-only property

File: src/Entities/Transactions/ThreeDSecureResult.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * @return string
+     */
+    public function getVersion()
+    {
+        return $this->getAttribute('version');
+    }
+
+    /**
+     * @return bool
```
Comment (ashkarpetin) [inline src/Entities/Transactions/ThreeDSecureResult.php:30]:
```suggestion
     * @return string
```

**Review by ashkarpetin — APPROVED** [review]

**Review by vykhrystiuk — APPROVED** [review]

**Comment by ashkarpetin** [comment]
Thanks!

**Pull request merged**